src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -639,6 +663,17 @@ internal static void SetupCallbacks(TraceEventDispatcher source)
 
                         // fire event
                         if (stats.GCStart != null) stats.GCStart(process, _gc);
+
+                        // check if we should apply a lifetime limit to the GC cache
+                        if (source.DataLifetimeMsec > 0 && data.TimeStampRelativeMSec >= stats.GC.NextRelativeTimeStampMsec)
+                        {
+                            // note the next time that lifetime should be applied, to avoid cleaningup too frequently
+                            stats.GC.NextRelativeTimeStampMsec = data.TimeStampRelativeMSec + (source.DataLifetimeMsec/2.0);
+                            // trim the GCs to only include those either incomplete or completed after lifetime
+                            stats.GC.m_gcs = stats.GC.m_gcs.Where(gc => !gc.IsComplete || gc.StartRelativeMSec >= (data.TimeStampRelativeMSec - source.DataLifetimeMsec)).ToList();
+                            // rewrite the index for fast lookup
+                            for (int i = 0; i < stats.GC.m_gcs.Count; i++) stats.GC.m_gcs[i].Index = i;
```

Comment:
nit, put the body on its own line.    I do this one line thing for some very short if conditions but I don't think it is a good idea for for loops because a for is longer.

Reply:
Done.